### PR TITLE
utils/scramble: scramble package

### PR DESCRIPTION
**Pull request by rvolosatovs**
scramble package provides various scrambling functions.
Is uses [crypto/rand] (https://golang.org/pkg/crypto/rand/) for random number generation.
used, for example, by the NOC - if users token does not grant access to gateway location, it gets scrambled, such that the real location does not get reported.

- [ ] A proper `delta` value should be determined for the gateway location field scrambling. What are your opinions?

### CODE REVIEW

File: utils/scramble/scramble.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
+package scramble
```
Comment (htdvisser) [inline utils/scramble/scramble.go]: Please add these to `utils/random` package

File: api/gateway/scramble.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package gateway
+
+import "github.com/TheThingsNetwork/ttn/utils/scramble"
+
+func (m *GPSMetadata) Scramble() (err error) {
```
Comment (htdvisser) [inline api/gateway/scramble.go]: Why are gateways in Northeast Russia scrambled over a larger distance than gateways on null island?

**Comment by htdvisser** [comment]
I don't think this code belongs in `TheThingsNetwork/ttn`. As far as I understand the functions you're adding are only needed in the NOC component. Why should they be implemented in `ttn`?

**Comment by rvolosatovs** [comment]
I am not sure actually. Thought it may be useful for `ttn` also.

File: api/gateway/scramble.go (hunk 2 above)
Comment (rvolosatovs) [inline api/gateway/scramble.go]: That's why I added the "A proper delta value should be determined for the gateway location field scrambling. What are your opinions?" point.
How, do you think, this should be done?

**Comment by rvolosatovs** [comment]
also, the `gateway.Location.Scramble()` seemed to be convenient and make sense

**Comment by johanstokking** [comment]
This should not be in `ttn`:

1. There is a trust relationship from the Router to its monitors anyway, since it forwards everything. There is no need to scramble anything before that, just trust the NOC to respect public/private status when exposing gateway status (location, counters etc)

2. Convenience code that we don't use should not be part of this codebase. `ttn` shall be lean and clean. Stuff for later use can be in branches, it really necessary

**Pull request closed without merging**